azure-pipelines.yml

```diff
@@ -128,7 +128,26 @@ stages:
           jdkVersionOption: '1.11'
           sonarQubeRunAnalysis: false
 
-  - stage: TESTS2
+  - stage: TESTS3
+    dependsOn: BUILD
+    displayName: Tests - Unit
+    jobs:
+    - job:
+      pool:
+        vmImage: 'windows-latest'
+      steps:
+      - task: Gradle@2
+        inputs:
+          gradleWrapperFile: 'gradlew'
+          workingDirectory: '$(System.DefaultWorkingDirectory)'
+          tasks: clean test -DTEST_SUBSCRIPTION_KEY=$(hmi-apim-svc-sbox-sub-key) -DTEST_URL=https://hmi-apim-svc-sbox.azure-api.net -DTEST_HOST=hmi-apim-svc-sbox.azure-api.net
```

Comment:
If this is the only bit different from functional and smoke, please modularise as template, doesn't look good to copy tens of lines of code for just one line.

Reply:
Also, can we stick to convention "clean functional" "clean smoke" and rename this to match, like "clean unit" to indicate what kind of test is running?

Reply:
I take your first point.
On second point I however I kind of disagree, as by convention the unit tests for a project reside in the test directory and the gradle task is named accordingly. 
So I would like to keep it as it is.